### ISSUE TITLE
Not-equal-to compare queries

### ISSUE THREAD

**Issue by threshold862543** [dao-xyz/peerbit#47]
Something that could be useful is adding `Compare.NotEqual` to `FieldBigIntCompareQuery,` to find all documents where the value is not equal to the specified value.

Something analogous could also be useful with `FieldStringMatchQuery`.

**Comment by marcus-pousette** [comment]
Yes this make sense. I start to think about whether queries should be able to be nested, like then you can wrap everthing in a "not", like

not( property x match "abc" and  property z < 123)

then you can negate and query, bot also negate the '$in'  #46
